### PR TITLE
API: enhance SearchIssues swagger docs

### PR DESCRIPTION
**Pull request by 6543**
this will result in better api clients generated out of the openapi docs ... for SearchIssues

---
*Sponsored by Kithara Software GmbH*

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: routers/api/v1/repo/issue.go [hunk 1]
```diff
@@ -41,80 +41,93 @@ func SearchIssues(ctx *context.APIContext) {
...
 	//   in: query
-	//   description: Only show notifications updated before the given time. This is a timestamp in RFC 3339 format
+	//   description: Only show issues updated before the given time (RFC 3339 format)
 	//   type: string
 	//   format: date-time
-	//   required: false
 	// - name: assigned
 	//   in: query
-	//   description: filter (issues / pulls) assigned to you, default is false
+	//   description: Filter issues assigned to the authenticated user
```
Comment (delvh) [inline routers/api/v1/repo/issue.go]: `or pulls`

File: routers/api/v1/repo/issue.go [hunk 2]
```diff
@@ -41,80 +41,93 @@ func SearchIssues(ctx *context.APIContext) {
...
 	//   type: boolean
+	//   default: false
 	// - name: created
 	//   in: query
-	//   description: filter (issues / pulls) created by you, default is false
+	//   description: Filter issues created by the authenticated user
```
Comment (delvh) [inline routers/api/v1/repo/issue.go]: See above

File: routers/api/v1/repo/issue.go [hunk 3]
```diff
@@ -41,80 +41,93 @@ func SearchIssues(ctx *context.APIContext) {
...
 	//   type: boolean
+	//   default: false
 	// - name: mentioned
 	//   in: query
-	//   description: filter (issues / pulls) mentioning you, default is false
+	//   description: Filter issues mentioning the authenticated user
```
Comment (delvh) [inline routers/api/v1/repo/issue.go]: See above

**Review by techknowlogick — APPROVED** [review]

**Comment by 6543** [comment]
-> #32298

**Pull request merged**